src/geo-objects/ObjectManager.js

```diff
@@ -56,6 +57,17 @@ class ObjectManager extends React.Component {
       instance.objects.options.set(objects || {})
     }
 
+    if (!isEqual(features, prevFeatures)) {
+      const prevIds = prevFeatures.map(feature => feature.id)
+      const currentIds = features.map(feature => feature.id)
+      const toBeRemovedIds = difference(prevIds, currentIds)
+      const toBeAddedIds = difference(currentIds, prevIds)
+      const toBeAddedObjects = features.filter(feature => toBeAddedIds.includes(feature.id))
+
+      instance.remove(toBeRemovedIds)
+      instance.add(toBeAddedObjects)
```

Comment:
I tried this on map with ~5k objects, no perf drops were noticed. (Moreover, map itself already wasn't smooth, but that feature replace operation was unnoticable) The propblem with your approach is when you remove all objects and add them again - they all will blink(tested it), and that is, I guess, annoying for user.